examples/experimental/dagster-airlift/dagster_airlift/in_airflow/base_proxy_operator.py

```diff
@@ -63,127 +98,225 @@ def _get_validated_session(self, context: Context) -> requests.Session:
-        return None
+
+    def get_valid_graphql_response(self, response: Response, key: str) -> Any:
+        response_json = response.json()
+        if not response_json.get("data"):
+            raise Exception(f"Error in GraphQL request. No data key: {response_json}")
+
+        if key not in response_json["data"]:
+            raise Exception(f"Error in GraphQL request. No {key} key: {response_json}")
+
+        return response_json["data"][key]
+
+    def _execute_proxied_dagster_runs(self, context: Context) -> None:
+        assets = self.get_assets_to_trigger()
+        triggered_runs = self.trigger_runs_for_assets(assets)
+        self.wait_for_runs_to_complete(triggered_runs)
 
     def execute(self, context: Context) -> Any:
         # https://github.com/apache/airflow/discussions/24463
         os.environ["NO_PROXY"] = "*"
-        dag_id = os.environ["AIRFLOW_CTX_DAG_ID"]
-        task_id = os.environ["AIRFLOW_CTX_TASK_ID"]
-        return self.launch_runs_for_task(context, dag_id, task_id)
+        self._execute_proxied_dagster_runs(context)
+
+
+def _group_assets_by_job_identifier(
+    assets: Sequence[ExecutableAssetIdentifier],
+) -> Mapping[JobIdentifier, Sequence[Sequence[str]]]:
+    assets_by_job_identifier: Dict[JobIdentifier, List[Sequence[str]]] = defaultdict(list)
+    for asset in assets:
+        assets_by_job_identifier[
+            (asset["location_name"], asset["repository_name"], asset["job_name"])
+        ].append(asset["asset_key_path"])
+    return assets_by_job_identifier
+
+
+def _build_execution_params(
+    tags: Mapping[str, str],
+    asset_key_paths: Sequence[Sequence[str]],
+    location_name: str,
+    repository_name: str,
+    job_name: str,
+) -> Dict[str, Any]:
+    return {
+        "mode": "default",
+        "executionMetadata": {
+            "tags": [{"key": key, "value": value} for key, value in tags.items()]
+        },
+        "runConfigData": "{}",
+        "selector": {
+            "repositoryLocationName": location_name,
+            "repositoryName": repository_name,
+            "pipelineName": job_name,
+            "assetSelection": [{"path": asset_key_path} for asset_key_path in asset_key_paths],
+            "assetCheckSelection": [],
+        },
+    }
+
 
+class DefaulProxyToDagsterOperator(BaseProxyToDagsterOperator):
+    """Underlying implementation of a DagsterOperator. This one uses the DAGSTER_URL environment variable to connect to the Dagster instance."""
 
-class DefaultProxyToDagsterOperator(BaseProxyToDagsterOperator):
-    def get_dagster_session(self, context: Context) -> requests.Session:
+    @lru_cache(maxsize=1)
+    def get_dagster_session(self) -> requests.Session:
         return requests.Session()
 
-    def get_dagster_url(self, context: Context) -> str:
+    def get_dagster_url(self) -> str:
         return os.environ["DAGSTER_URL"]
 
 
+class ProxiedTaskDagsterOperator(DefaulProxyToDagsterOperator):
```

Comment:
I think this name is much more clear, and the small implementation surface area is particularly nice.